packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -272,16 +273,16 @@ library LibProposing {
 
     /// @dev Hashing the block metadata.
     function _hashMetadata(TaikoData.BlockMetadata memory meta)
-        private
+        internal
         pure
         returns (bytes32 hash)
     {
         uint256[5] memory inputs;
         inputs[0] = uint256(meta.l1Hash);
         inputs[1] = uint256(meta.mixHash);
         inputs[2] = uint256(meta.txListHash);
-        inputs[3] = (uint256(meta.timestamp) << 192)
-            | (uint256(meta.l1Height) << 128) | (uint256(meta.gasLimit) << 96);
+        inputs[3] = (uint256(meta.id)) | (uint256(meta.timestamp) << 64)
+            | (uint256(meta.l1Height) << 128) | (uint256(meta.gasLimit) << 192);
```

Comment:
`(uint256(meta.gasLimit) << 192` to `(uint256(meta.gasLimit) << 160`? as gasLimit is a uint32

Reply:
Yes, because 128 is position where the `l1Height` starts and since `l1Height` is 64, so `gasLimit` can only start at 192.